### ISSUE TITLE
Improve documentation around using remote backends such as Consul

### ISSUE THREAD

**Issue by devcamcar** [hashicorp/terraform#891]
I am currently trying to figure out how to get Terraform's Consul remote to function, but I've found it confusing to get working.  For instance, I try:

    terraform remote -backend=consul -address=consul.mydomain.com:8500 -path=terraform
    terraform plan

    Refreshing Terraform state prior to plan...

    Error refreshing state: Get /: unsupported protocol scheme ""

So then I try:

    terraform remote -backend=consul -address=http://consul.mydomain.com:8500 -path=terraform
    terraform plan
    Failed to refresh state: Failed to read remote state: Get http://https://consul.nebula.com:8500/v1/kv/terraform/devcamcar: dial tcp: too many colons in address https://consul.mydomain.com:8500

Okay... so then I try:

    terraform remote -backend=consul -address=http://consul.mydomain.com -path=terraform
    terraform plan
    Failed to refresh state: Failed to read remote state: Get http://consul.mydomain.com/v1/kv/terraform: dial tcp 10.[redacted]:80: connection refused

So basically the parser won't let me add the port number because it thinks there are too many colons.  Also worth noting if you don't specify -address at all, then you get this, which shows the url (with localhost replaced obviously) as having http:// and :8500.

    terraform remote -backend=consul -path=terraform
    terraform apply
    Failed to refresh state: Failed to read remote state: Get http://127.0.0.1:8500/v1/kv/terraform: dial tcp 127.0.0.1:8500: connection refused

**Comment by devcamcar** [comment]
Figured out that the error "Error refreshing state: Get /: unsupported protocol scheme" was really coming from an experimental OpenStack provider. Closing this ticket.

**Comment by strarsis** [comment]
Thanks, this was very helpful for me as I left the token just empty (for dev) 
and got the same error message with consul-template.
